core/src/main/java/org/apache/calcite/interpreter/Bindables.java

```diff
@@ -836,6 +836,11 @@ public static class BindableWindow extends Window implements BindableRel {
           constants, getRowType(), groups);
     }
 
+    @Override public Window copy(List<RexLiteral> constants) {
```

Comment:
There should be a comment that indicates that this _replaces_ (not adds) the existing constants on the node with the supplied constants.

Reply:
Added a Javadoc comment in the `Window` class.